ale/drivers/kaguya_driver.py

```diff
@@ -30,21 +42,28 @@ def metakernel(self):
                     self._metakernel = mk
         return self._metakernel
 
+    @property
+    def ending_ephemeris_time(self):
+        if not hasattr(self, '_ending_ephemeris_time'):
+            self._ending_ephemeris_time = self.label.get('CORRECTED_SC_CLOCK_STOP_COUNT').value
+            self._ending_ephemeris_time = spice.sct2e(self.spacecraft_id, self._ending_ephemeris_time)
+        return self._ending_ephemeris_time
+
+
```

Comment:
I would assume the only way the hasattr(self, '_ending_ephemeris_time') would return true is if the base.py driver's method was successful in deriving '_ending_ephemeris_time'. However, base.py defines 

"
@property
    def ending_ephemeris_time(self):
        return (self.image_lines * self.line_exposure_duration) + self.starting_ephemeris_time
"

the property is defined 'ending_ephemeris_time', without the preceding '_' and the return is a function, not 'self._ending_ephemeris_time', so this check may always be false.

Reply:
It has nothing to do with deriving. That statement is always true the first time you call it. It immediately adds `_ending_ephemeris_time` as an attribute after the first call, so then it will be false from then on. It is caching the result, so you don't hit spice constantly. I'm more for removing most of these since it only makes sense for IO heavy operations (we wouldn't want to keep reading a potentially large table fo data, but a single number? Who cares.).